src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -0,0 +1,105 @@
+                    Interop.libcurl.CURLoption.CURLOPT_SSL_CTX_FUNCTION,
+                    s_sslCtxCallback);
+
+                switch (answer)
+                {
+                    case Interop.libcurl.CURLcode.CURLE_OK:
+                        break;
+                    case Interop.libcurl.CURLcode.CURLE_NOT_BUILT_IN:
+                        VerboseTrace("CURLOPT_SSL_CTX_FUNCTION is not supported, platform default https chain building in use");
+                        break;
+                    default:
+                        ThrowIfCURLEError(answer);
+                        break;
+                }
+            }
+
+            private static int SetSslCtxVerifyCallback(
+                IntPtr curl,
+                IntPtr sslCtx,
+                IntPtr userPtr)
+            {
+                using (SafeSslContextHandle ctx = new SafeSslContextHandle(sslCtx, ownsHandle: false))
+                {
+                    Interop.Ssl.SslCtxSetCertVerifyCallback(ctx, s_sslVerifyCallback, userPtr);
+                }
+
+                return Interop.libcurl.CURLcode.CURLE_OK;
+            }
+
+            private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
+            {
+                List<X509Certificate2> otherCerts;
+                bool success;
+
+                using (SafeX509StoreCtxHandle storeCtx = new SafeX509StoreCtxHandle(storeCtxPtr, ownsHandle: false))
+                using (X509Chain chain = new X509Chain())
+                {
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+
+                    IntPtr certPtr = Interop.Crypto.X509StoreCtxGetTargetCert(storeCtx);
+
+                    if (IntPtr.Zero == certPtr)
+                    {
+                        Debug.Fail("Invalid target certificate");
+                        return -1;
+                    }
+
+                    using (SafeSharedX509StackHandle extraStack = Interop.Crypto.X509StoreCtxGetSharedUntrusted(storeCtx))
+                    {
+                        int extraSize = extraStack.IsInvalid ? 0 : Interop.Crypto.GetX509StackFieldCount(extraStack);
+                        otherCerts = new List<X509Certificate2>(extraSize);
+
+                        for (int i = 0; i < extraSize; i++)
+                        {
+                            certPtr = Interop.Crypto.GetX509StackField(extraStack, i);
+
+                            if (certPtr != IntPtr.Zero)
+                            {
+                                X509Certificate2 cert = new X509Certificate2(certPtr);
+                                otherCerts.Add(cert);
+                                chain.ChainPolicy.ExtraStore.Add(cert);
+                            }
+                        }
+                    }
+
+                    using (X509Certificate2 leafCert = new X509Certificate2(certPtr))
```

Comment:
Oh, wait, I see now.  I reused the variable... and lost the value I cared about (I guess I restructured the body of this too many times).  The thing that made me realize that was a message that I think you deleted :)